.gitlab-ci.yml

```diff
@@ -5,23 +5,33 @@ stages:
   - deploy
 
 .js:
-  image: node
+  image: node:15 # Keep version in sync with franklin
   before_script:
     - git --version
     - node --version
     - npm --version
     - yarn --version
-    - yarn
 
+# TODO: when we want to run test:functional and test:visual use image: buildkite/puppeteer
 test:
-  image: buildkite/puppeteer
+  extends: .js
   stage: test
   script:
-    - apt-get update && apt-get install -y git
-    - git --version
-    - node --version
-    - npm --version
-    - yarn --version
+    - echo $CI_PIPELINE_SOURCE $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME $CI_MERGE_REQUEST_TARGET_BRANCH_NAME
+    - yarn
+    - yarn test
+
+merge_and_test:
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "external_pull_request_event"
+      when: always
+  extends: .js
+  stage: test
+  script:
+    - echo Merging $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME into $CI_MERGE_REQUEST_TARGET_BRANCH_NAME to test upstream effects
+    - git checkout $CI_MERGE_REQUEST_TARGET_BRANCH_NAME
+    - git merge $CI_MERGE_REQUEST_SOURCE_BRANCH_NAME -m "Merge and test"
+    - test -n "$(git status --porcelain)" && exit 1 # Exit if merge has conflicts
     - yarn
     - yarn test
```

Comment:
Don't you need to add this somewhere:
```
only:
    - external_pull_requests
```

Reply:
Was trying out [rules](https://docs.gitlab.com/ee/ci/merge_request_pipelines/#use-rules-to-run-pipelines-for-merge-requests) but if this doesn't work then we should try your suggestion.